services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -495,10 +534,22 @@ async def manage_get_response(
         if common_settings.CLUSTER_MODE:  # Get the list of access and the user_login calling the endpoint.
             auth_roles = request.state.auth_roles
             user_login = request.state.user_login
-        if request.scope["path"] == "/" and (common_settings.CLUSTER_MODE):  # /catalog and /catalog/catalogs/owner_id
-            content = manage_landing_page(request, auth_roles, user_login, content, user)
-            if hasattr(content, "status_code"):  # Unauthorized
-                return content
+        if request.scope["path"] == "/":
+            if common_settings.CLUSTER_MODE:  # /catalog and /catalog/catalogs/owner_id
+                content = manage_landing_page(request, auth_roles, user_login, content, user)
+                if hasattr(content, "status_code"):  # Unauthorized
+                    return content
+            # Manage local landing page of the catalog
+            regex_catalog = r"/collections/(?P<owner_id>.+?)_(?P<collection_id>.*)"
+            for i, link in enumerate(content["links"]):
```

Comment:
In python is suggested to avoid index-based modification (content['links'][i] = * )

for link in content["links"]:
    if link["rel"] == "data":
        collection_suffix = "/collections"
        link["href"] = link["href"][: -len(collection_suffix)] + "/catalog/collections"
    
    link_parser = urlparse(link["href"])
    
    if match := re.match(regex_catalog, link_parser.path):
        groups = match.groupdict()
        new_path = add_user_prefix(link_parser.path, groups["owner_id"], groups["collection_id"])
        link["href"] = link_parser._replace(path=new_path).geturl()